Scripts/Player/States/death.gd

```
@@ -0,0 +1,30 @@
+class_name PlayerDeath
+extends PlayerState
+
+var sprite: AnimatedSprite2D
+var death_timer: float = 0.0
+var death_duration: float = 1.5
+
+signal death_animation_complete
+
+func Enter():
+	if player:
```

same as in another comment about the skeleton, no need to check for player, if it's falsy then we messed up big time and having the game crash while testing is gonna be better than having to find out what silently failed and where. In this case in particular, the player states all have access to the player node as long as we connect the player variable using the UI. If we want to do it programmatically to reduce the amount of manual setup and avoid potential forgetfullness it's feasible too on the base PlayerState class (and create a base state class for each enemy), that's perhaps an ideal solution to greatly reduce boilerplate